src/RewardsStreamerMP.sol

```diff
@@ -131,14 +130,37 @@ contract RewardsStreamerMP is ReentrancyGuard {
         totalMP -= mpToReduce;
         potentialMP -= potentialMPToReduce;
 
-        bool success = STAKING_TOKEN.transfer(msg.sender, amount);
+        bool success = STAKE_TOKEN.transfer(msg.sender, _amount);
         if (!success) {
-            revert StakingManager__TransferFailed();
+            revert StakeManager__TransferFailed();
         }
 
         user.userRewardIndex = rewardIndex;
     }
 
+    function lock(uint256 _secondsIncrease) external onlyTrustedCodehash {
+        //TODO: increase lock time
+        revert("Not implemented");
+    }
+
+    function exit() external returns (bool _leaveAccepted) {
+        if (!isTrustedCodehash(msg.sender.codehash)) {
+            //case owner removed access from a class of StakeVault,. they might exit
+            delete user[msg.sender];
+            return true;
+        } else {
+            //TODO: handle other cases
+            //TODO: handle update/migration case
+            //TODO: handle emergency exit
+            revert("Not implemented");
+        }
+    }
+
+    function acceptUpdate() external onlyTrustedCodehash returns (address _migrated) {
+        //TODO: handle update/migration
+        revert("Not implemented");
+    }
```

Comment:
As mentioned above, let's tackle that in #20

Reply:
Ok, we can leave it as Not Implemented here for now and later we implement the logic #20

Reply:
Sounds good to me, but are we sure we'll end up with an `acceptUpdate` function?
If yes, then it's fine

Reply:
If we want to keep compatibility for StakeVault with both systems, yes, but for this contact, acceptUpdate might be just a `return address(this)`, if users don't need to accept updates. But that discussion is far more complex, as a Proxy Update means a breach of contract, and morally it should need to accept the change - however I don't see other developers caring about the morals of contract breaching.

Reply:
> If we want to keep compatibility for StakeVault with both systems, yes, 

This is not a priority atm.